specification/cosmos-db/resource-manager/Microsoft.DocumentDB/preview/2023-03-01-preview/mongoCluster.json

```diff
@@ -0,0 +1,983 @@
+          },
+          "description": "Application-specific metadata in the form of key-value pairs."
+        }
+      }
+    },
+    "MongoClusterProperties": {
+      "type": "object",
+      "description": "The properties of a mongo cluster.",
+      "properties": {
+        "createMode": {
+          "type": "string",
+          "description": "The mode to create a mongo cluster.",
+          "enum": [
+            "Default",
+            "PointInTimeRestore"
+          ],
+          "x-ms-enum": {
+            "name": "CreateMode",
+            "modelAsString": true
+          },
+          "x-ms-mutability": [
+            "create"
+          ],
+          "default": "Default"
+        },
+        "restoreParameters": {
+          "$ref": "#/definitions/MongoClusterRestoreParameters",
+          "description": "Parameters used for restore operations",
+          "x-ms-mutability": [
+            "create"
+          ]
+        },
+        "administratorLogin": {
+          "type": "string",
+          "description": "The administrator's login for the mongo cluster.",
+          "x-ms-mutability": [
+            "create",
+            "update",
+            "read"
+          ]
+        },
+        "administratorLoginPassword": {
+          "type": "string",
+          "x-ms-secret": true,
+          "format": "password",
+          "x-ms-mutability": [
+            "create",
+            "update"
+          ],
+          "description": "The password of the administrator login."
+        },
+        "serverVersion": {
+          "type": "string",
+          "description": "The Mongo DB server version. Defaults to the latest available version if not specified."
+        },
+        "connectionString": {
+          "type": "string",
+          "readOnly": true,
+          "description": "The default mongo connection string for the cluster."
+        },
+        "earliestRestoreTime": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Earliest restore timestamp in UTC ISO8601 format."
+        },
+        "provisioningState": {
+          "$ref": "#/definitions/MongoClusterProvisioningState",
+          "description": "A provisioning state of the mongo cluster.",
+          "readOnly": true
+        },
+        "clusterStatus": {
+          "$ref": "#/definitions/MongoClusterStatus",
+          "description": "A status of the mongo cluster."
+        },
+        "nodeGroupSpecs": {
+          "$ref": "#/definitions/NodeGroupSpecList",
+          "description": "The list of node group specs in the cluster."
+        }
+      }
+    },
+    "NodeGroupSpecList": {
+      "type": "array",
+      "description": "The list of node group specifications for the cluster. Must include one node group spec with kind = 'Shard'.",
+      "items": {
+        "$ref": "#/definitions/NodeGroupSpec"
+      },
+      "x-ms-identifiers": [],
+      "minItems": 1,
+      "maxItems": 1,
+      "uniqueItems": true
+    },
```

Comment:
I think my main concern with having these limits declared in this apiVersion may 'fool' people into thinking they apply in future versions when that may not necessarily be the case?
Or if it were the case, why is it a list at all?

Reply:
Fair - we can remove the max limit here.